protocol_decoder/src/decoding.rs

```diff
@@ -500,16 +542,10 @@ fn create_dummy_gen_input(
     }
 }
 
-impl TxnMetaState {
-    fn txn_bytes(&self) -> Vec<u8> {
-        match self.txn_bytes.as_ref() {
-            Some(v) => v.clone(),
-            None => Vec::default(),
-        }
-    }
-}
-
-fn create_dummy_proof_trie_inputs(final_trie_state: &PartialTrieState) -> TrieInputs {
+fn create_dummy_proof_trie_inputs(
+    final_trie_state: &PartialTrieState,
+    state_trie: HashedPartialTrie,
+) -> TrieInputs {
```

Comment:
The namings here aren't straightforward. We need a massive documentation effort anyway across the decoder (proof-gen documentation is somehow acceptable now). But to not increase complexity once we merge this branch fix to `main` I'd suggest documenting the newest changes for clarity.